phpunit/tests/fonts/font-library/wpFontCollection/getData.php

```diff
@@ -12,13 +12,50 @@
  */
 class Tests_Fonts_WpFontCollection_GetData extends WP_UnitTestCase {
 
+	public function set_up() {
+		parent::set_up();
+
+		// Mock the wp_remote_request() function.
+		add_filter( 'pre_http_request', array( $this, 'mock_request' ), 10, 3 );
+	}
+
+	public function tear_down() {
+		// Remove the mock to not affect other tests.
+		remove_filter( 'pre_http_request', array( $this, 'mock_request' ) );
+
+		parent::tear_down();
+	}
+
+	public function mock_request( $preempt, $args, $url ) {
+		// Check if it's the URL you want to mock.
+		if ( 'https://localhost/fonts/mock-font-collection.json' === $url ) {
+
+			// Mock the response body.
+			$mock_collection_data = array(
+				'fontFamilies' => 'mock',
+				'categories'   => 'mock',
+			);
+
+			return array(
+				'body'     => json_encode( $mock_collection_data ),
+				'response' => array(
+					'code' => 200,
+				),
+			);
+		}
+
+		// For any other URL, return false which ensures the request is made as usual (or you can return other mock data).
+		return false;
+	}
```

Comment:
I suggest using return early pattern to avoid nesting: https://github.com/piotrplenik/clean-code-php#avoid-nesting-too-deeply-and-return-early-part-1
```suggestion
	public function mock_request( $preempt, $args, $url ) {
		// if the URL is not the URL you want to mock, return false.
		if ( 'https://localhost/fonts/mock-font-collection.json' !== $url ) {
			return false;
		}

		// Mock the response body.
		$mock_collection_data = array(
			'fontFamilies' => 'mock',
			'categories'   => 'mock',
		);

		return array(
			'body'     => json_encode( $mock_collection_data ),
			'response' => array(
				'code' => 200,
			),
		);
	}
```